### PR TITLE
fix: cleaning x-space related tests

### PR DESCRIPTION
**Pull request by Cyberxon**
fixing the cross space references integration tests

### CODE REVIEW

File: test/integration/entry-integration.js [hunk 1]
```diff
@@ -480,48 +474,76 @@ describe('Entry Api', () => {
...
-        'Testing Environment'
-      )
-      xSpaceDisabledContentType = await xSpaceDisabledEnvironment.createContentTypeWithId(
-        generateRandomId('test-content-type'),
-        contentTypeData
-      )
-      await xSpaceDisabledContentType.publish()
+      test('can create, publish, unpublish and delete an entry with xspace references', async () => {
+        return xSpaceEnabledEnvironment
+          .createEntry(xSpaceEnabledContentType.sys.id, entryData)
```
Comment (marcolink) [inline test/integration/entry-integration.js:493]: Please do not mutate the default space here! 
If you need to mutate, clone the environment using `createTestEnvironment` and do any write operation exclusively on this environment.

**Review by marcolink — CHANGES_REQUESTED** [review]

File: test/integration/entry-integration.js (hunk 1 above)
Comment (Cyberxon) [reply test/integration/entry-integration.js:493]: Just to explain, I needed a space with cross space feature enabled and enabling it for a space needs additional steps like adding the `space_id` in the feature flags dedicated for the feature. Your idea of cloning master worked for me so I am pushing the newest changes.

File: test/integration/entry-integration.js [hunk 2]
```diff
@@ -480,48 +476,87 @@ describe('Entry Api', () => {
...
+      let xSpaceDisabledEnvironment
+      let xSpaceDisabledContentType
+      let xSpaceDisabledEntryId
 
-      await xSpaceDisabledSpace.delete()
+      before(async () => {
+        // Creating a new space that has the x-space feature disabled
+        xSpaceDisabledSpace = await createTestSpace(initClient(), 'XSpaceDisabledEntry')
+        xSpaceDisabledEnvironment = await xSpaceDisabledSpace.getEnvironment('master')
+        await waitForEnvironmentToBeReady(xSpaceDisabledSpace, xSpaceDisabledEnvironment)
```
Comment (marcolink) [inline test/integration/entry-integration.js:530]: `await createTestSpace` should return a fully created space with a ready `master` environment. 
I would expect that you don't necessarily have to wait for the `master` environment to b ready.

File: test/integration/entry-integration.js (hunk 1 above)
Comment (marcolink) [reply test/integration/entry-integration.js:505]: Thanks for the explanation @Cyberxon 🙏 
I think that's all totally fine, and the latest changes reflect the best possible test setup for it :)

**Review by marcolink — APPROVED** [review]
Looks great - thanks for fixing it 🎉

**Pull request merged**